### PR TITLE
Added selectorsyncset generation

### PR DESCRIPTION
**Pull request by rafael-azevedo**
This PR added a script and template to generate selectorsyncset. In addition it added a Makefile and project.mk to run the script and pass in the required parameters.

### CODE REVIEW

**Comment by rafael-azevedo** [comment]
We will need to create a directory as required for hive to pull the yaml file and then change the variable SELECTOR_SYNC_SET_DESTINATION in projects.mk to reflect this directory change.

File: scripts/generate_syncset.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    for file in file_paths:
+        objects = get_yaml_all(file)
+        for obj in objects:
+            yaml_objs.append(obj)
+    return yaml_objs
+
+
+if __name__ == '__main__':
+    #Argument parser
+    parser = argparse.ArgumentParser(description="selectorsyncset generation tool")
```
Comment (jharrington22) [inline scripts/generate_syncset.py]: Adding `, usage='%(prog)s [options]'` to `argparse.ArgumentParser()` would print the name of the file in the help usage like `usage: generate_syncset.py [options]`

File: scripts/templates/selectorsyncset.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+apiVersion: hive.openshift.io/v1alpha1
+kind: SelectorSyncSet
+metadata:
+  generation: 1
+  name: osd-manged-cluster-config
+  labels:
+    osd-config: True
```
Comment (jharrington22) [inline scripts/templates/selectorsyncset.yaml]: @jewzaam @rafael-azevedo should we make this osdConfig, looks kind of odd but matches convention.

File: scripts/generate_syncset.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+def get_yaml(filename):
+    with open(filename,'r') as input_file:
+        return yaml.safe_load(input_file)
+
+def get_all_yaml_files(path):
+    file_paths = []
+    for r,_,f in os.walk(path):
+        for file in f:
+            if '.yml' in file:
```
[the remark this answers is not in the document]
Comment (jewzaam) [reply scripts/generate_syncset.py]: And we want files that end with those extensions, not have them in the middle.

File: scripts/templates/selectorsyncset.yaml (hunk 2 above)
Comment (jewzaam) [inline scripts/templates/selectorsyncset.yaml]:
```suggestion
    managed.openshift.io/osd: "true"
```

Comment (jewzaam) [reply scripts/templates/selectorsyncset.yaml]: Also were we not going to put the URL to the git repo?  Hash is going to be a little bit of a pain to put in, but would be nice.  Not required.

```
    managed.openshift.io/git-source: "https://github.com/openshift/managed-cluster-config/"
```

Comment (rafael-azevedo) [reply scripts/templates/selectorsyncset.yaml]: is `managed.openshift.io/osd: "true"` to generic ? Maybe `managed.openshift.io/osd-config: "true"`

Comment (rafael-azevedo) [reply scripts/templates/selectorsyncset.yaml]: This is resolved by the method in the comments below.

Comment (rafael-azevedo) [reply scripts/templates/selectorsyncset.yaml]: Created the labels pr our conversation

**Comment by jharrington22** [comment]
@jewzaam I still feel the following is better as it conforms to current Kubernetes key names.

```
  labels:
    managed.openshift.io/osd: True 
    managed.openshift.io/gitSource: 
    managed.openshift.io/gitHash:
```

Thoughts?

**Comment by rafael-azevedo** [comment]
> @jewzaam I still feel the following is better as it conforms to current Kubernetes key names.
> 
> ```
>   labels:
>     managed.openshift.io/osd: True 
>     managed.openshift.io/gitSource: 
>     managed.openshift.io/gitHash:
> ```
> 
> Thoughts?

I agree, but I think either will work.

**Comment by jewzaam** [comment]
> > @jewzaam I still feel the following is better as it conforms to current Kubernetes key names.
> > ```
> >   labels:
> >     managed.openshift.io/osd: True 
> >     managed.openshift.io/gitSource: 
> >     managed.openshift.io/gitHash:
> > ```
> > 
> > 
> > Thoughts?
> 
> I agree, but I think either will work.

That'll be fine for the key.  Taking a look at what is allowed for label values we'll not be able to put the git repo url into gitSource.  See https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set: `Valid label values must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.`

**Comment by jharrington22** [comment]
@jewzaam Interesting I didn't know there was a character restriction or limit.

**Comment by rafael-azevedo** [comment]
> > > @jewzaam I still feel the following is better as it conforms to current Kubernetes key names.
> > > ```
> > >   labels:
> > >     managed.openshift.io/osd: True 
> > >     managed.openshift.io/gitSource: 
> > >     managed.openshift.io/gitHash:
> > > ```
> > > 
> > > 
> > > Thoughts?
> > 
> > 
> > I agree, but I think either will work.
> 
> That'll be fine for the key. Taking a look at what is allowed for label values we'll not be able to put the git repo url into gitSource. See https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set: `Valid label values must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.`

We can shorten the url in the python code.

File: scripts/templates/selectorsyncset.yaml [hunk 4]
```diff
@@ -4,13 +4,8 @@ kind: SelectorSyncSet
 metadata:
   generation: 1
   name: osd-manged-cluster-config
-  labels:
-    managed.openshift.io/osd: True
```
Comment (jewzaam) [inline scripts/templates/selectorsyncset.yaml]: Hive is using all strings. Not sure about reason. We are not (at this time) planning to do anything with these other than help in operations and reporting spits, probably fine.

**Comment by jewzaam** [comment]
I see the URL is removed. Decided shortening won't work or we don't really need the URL?

**Comment by rafael-azevedo** [comment]
I decided that both shortening didn't work and we probably didn't need it.

**Pull request merged**